pkg/daemon/update.go

```diff
@@ -732,7 +979,48 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig, skipCertifi
 	logSystem("Starting update from %s to %s: %+v", oldConfigName, newConfigName, diff)
 
 	diffFileSet := ctrlcommon.CalculateConfigFileDiffs(&oldIgnConfig, &newIgnConfig)
-	actions, err := calculatePostConfigChangeAction(diff, diffFileSet)
+	diffUnitSet := ctrlcommon.CalculateConfigUnitDiffs(&oldIgnConfig, &newIgnConfig)
+
+	var fg featuregates.FeatureGate
+
+	// This check is needed as featureGatesAccessor is not present during first boot. During firstboot
+	// the daemon will always do a reboot and NodeDisruptionPolicies are not active.
+	if dn.featureGatesAccessor != nil {
+		fg, err = dn.featureGatesAccessor.CurrentFeatureGates()
+		if err != nil {
+			klog.Errorf("Could not get fg: %v", err)
+			return err
+		}
+	}
+
+	var nodeDisruptionActions []opv1.NodeDisruptionPolicyStatusAction
+	var actions []string
+	// If FeatureGateNodeDisruptionPolicy is set, calculate NodeDisruptionPolicy based actions for this MC diff
+	if fg != nil && fg.Enabled(v1.FeatureGateNodeDisruptionPolicy) {
+		mcop, err := dn.mcopClient.OperatorV1().MachineConfigurations().Get(context.TODO(), "cluster", metav1.GetOptions{})
+		if mcop == nil || apierrors.IsNotFound(err) {
+			return err
```

Comment:
I'm wondering if in general we should have the backup (existing) path exist always just as a fallback in case the object doesn't exist for some reason.

For example, hypershift implementation has non-rebooting updates too but they probably won't have this object

Since this is tech preview and behind a FG I don't think we need to consider it now, but probably good to revisit for GA

Reply:
Agree with making a card to fallback for GA. I'd guess a user would want to know how it failed and why? I guess we could log it somehow that we are in the fallback path, but maybe we can have this go up the object status when we get that working

Reply:
Made a card for this: https://issues.redhat.com/browse/MCO-1146